src/lib/github/github-contributors.ts

```diff
@@ -0,0 +1,175 @@
+import { GithubTarget, Username, Contributor, ContributorMap } from '../types';
+import { Commits, Repo, Org } from './types';
+import { fetchAllPages } from './utils';
+
+import * as debugLib from 'debug';
+const debug = debugLib('snyk:github-count');
+const githubDefaultUrl = 'https://api.github.com/';
+
+export const fetchGithubContributors = async (
+  githubInfo: GithubTarget,
+  threeMonthsDate: string,
+): Promise<ContributorMap> => {
+  const contributorsMap = new Map<Username, Contributor>();
+  try {
+    let repoList: Repo[] = [];
+    // let orgList: Org[] = [];
+    if (githubInfo.repo && (!githubInfo.orgs || githubInfo.orgs.length > 1)) {
+      // If repo is specified, then a single project key is expected, bail otherwise
+      console.log(
+        'You must provide a single org name for single repo counting',
+      );
+      process.exit(1);
+    } else if (githubInfo.repo && githubInfo.orgs) {
+      // If repo is specified, and we got a single project key
+      debug('Counting contributors for single repo');
+      repoList.push({
+        name: githubInfo.repo,
+        owner: { login: githubInfo.orgs[0] },
+      });
+    } else if (!githubInfo.orgs) {
+      // console.log('You must provide at least one organization');
+      // process.exit(1);
+      githubInfo.orgs = [];
+      let orgList: Org[] = [];
+      orgList = orgList.concat(
+        await fetchGithubOrgs(
+          githubDefaultUrl + 'user/orgs?per_page=100',
+          githubInfo.token,
+        ),
+      ) as Org[];
+      for (let i = 0; i < orgList.length; i++) {
+        githubInfo.orgs.push(orgList[i].login);
+      }
+      repoList = repoList.concat(await fetchGithubReposForOrgs(githubInfo));
+    } else {
+      // Otherwise retrieve all repos (for given projects or all repos)
+      repoList = repoList.concat(await fetchGithubReposForOrgs(githubInfo));
```

Comment:
Same note here, take this out of the if else since it's the same?

Reply:
It breaks the script. It's similar but not exactly the same since if we provide a specific repo, it will still go and get all the repos..
I can add if(no specific org was given) but it's the same result. leaving it be for now.